platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/internal/format/EpochCachedFormatter.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.logging.api.internal.format;
+
+import static java.time.ZoneOffset.UTC;
+import static java.time.temporal.ChronoField.DAY_OF_MONTH;
+import static java.time.temporal.ChronoField.HOUR_OF_DAY;
+import static java.time.temporal.ChronoField.MINUTE_OF_HOUR;
+import static java.time.temporal.ChronoField.MONTH_OF_YEAR;
+import static java.time.temporal.ChronoField.NANO_OF_SECOND;
+import static java.time.temporal.ChronoField.SECOND_OF_MINUTE;
+import static java.time.temporal.ChronoField.YEAR;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.time.Instant;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.SignStyle;
+import java.time.temporal.ChronoUnit;
+import java.time.temporal.TemporalAccessor;
+import java.util.Map;
+import java.util.stream.IntStream;
+
+/**
+ * An epoc millis parser to human-readable String based on pattern: {@code "yyyy-MM-dd HH:mm:ss.SSS"}
+ */
+public class EpochCachedFormatter {
+
+    /**
+     * The formatter for the timestamp.
+     */
+    private static final DateTimeFormatter FORMATTER = new DateTimeFormatterBuilder()
+            .appendValue(YEAR, 4, 4, SignStyle.EXCEEDS_PAD)
+            .appendLiteral('-')
+            .appendValue(MONTH_OF_YEAR, 2)
+            .appendLiteral('-')
+            .appendValue(DAY_OF_MONTH, 2)
+            .appendLiteral(' ')
```

Comment:
@jsync-swirlds , I didn't quite get what you mean by _changes_ could you expand?
The Javadoc on the class states that the format the class produces is `yyyy-MM-dd HH:mm:ss.SSS`  and it doesn't refer to ISO-8601.
This is consistent with the current configuration for log4j we found in files in the repository.
Am I missing something?

Reply:
The prior implementation was ISO-8601 (with a `T` in the middle) because it used the `DateTimeFormatter.ISO_LOCAL_DATE_TIME` constant.
This PR changes that to a ` `.  I don't know if anything actually looks at dates in our logs, but this will change what we're producing, so I asked if we might want to publish that a little more widely.

Reply:
Oh! Of course, sorry I didn't get your point.
To add some context, this change has no production impact; we are still using log4j, but the previous implementation submitted used the wrong format. A look inside our repo revealed that the actual pattern in use in log4j coincides with the new one in the pull request.
![image](https://github.com/hashgraph/hedera-services/assets/152629744/7b3c4bc8-3e9e-4074-b482-24431620d0cd)